tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/trex/TransitionExecutorImpl.scala

```diff
@@ -0,0 +1,388 @@
+package at.forsyte.apalache.tla.bmcmt.trex
+
+import at.forsyte.apalache.tla.bmcmt._
+import at.forsyte.apalache.tla.bmcmt.rules.aux.{CherryPick, MockOracle, Oracle, SparseOracle}
+import at.forsyte.apalache.tla.bmcmt.util.TlaExUtil
+import at.forsyte.apalache.tla.lir.TlaEx
+import com.typesafe.scalalogging.LazyLogging
+
+/**
+  * <p>A general-purpose symbolic transition executor (or T-Rex).
+  * It accumulates the basic logic for executing TLA+
+  * transitions, which is used both in the sequential and parallel model checkers.
+  * (It could be used to implement predicate abstraction too.)
+  * This class is imperative, as taking SMT snapshots and recovering them
+  * in the non-incremental case is far from being functional.</p>
+  *
+  * <p>This class is parameterized by the type of an executor context. Currently, there are two choices:
+  * (1) IncrementalExecutorContext and (2) OfflineExecutorContext.</p>
+  *
```

Comment:
Do we need to duplicate the doc comments? Would be nice if we could just have them written in one place. E.g., perhaps this could just refer the reader to the documentation on the trait?

Reply:
Right. I think Intellij Idea is copy-pasting the docs. Will fix that.